edward/models/dirichlet_process.py

```diff
@@ -51,18 +52,25 @@ def __init__(self, alpha, base_cls, validate_args=False, allow_nan_stats=True,
 
         # Instantiate base distribution.
         self._base = self._base_cls(*self._base_args, **self._base_kwargs)
-        # Define atoms of Dirichlet process, storing only the first as default.
-        self._theta = tf.expand_dims(
-            self._base.sample(self.get_batch_shape()), 0)
+        # Create empty tensor to store future atoms
+        self._theta = tf.zeros(
+            [0] +
+            self.get_batch_shape().as_list() +
+            self.get_event_shape().as_list())
+
+        # Instantiate beta distribution for stick breaking proportions
+        self._betadist = Beta(a=tf.ones_like(self.alpha), b=self.alpha)
+        # Create empty tensor to store stick breaking proportions
+        self._beta = tf.zeros([0] + self.get_batch_shape().as_list())
 
         super(DirichletProcess, self).__init__(
-            dtype=tf.int32,
+            dtype=tf.float32,
```

Comment:
I could have the wrong end of the stick here but, isn't the process associated with the thetas? They don't have int32 type. Perhaps this should be `base_cls.dtype`